apps/site/src/app/(main)/portal/@applicant/ApplicantPortal.tsx

```diff
@@ -21,12 +21,18 @@ export const enum PortalStatus {
 	void = "VOID",
 }
 
-async function Portal() {
+async function Portal({
+	searchParams,
+}: {
+	searchParams?: {
+		return_to?: string;
+	};
+}) {
 	const identity = await getUserIdentity();
 	const status = identity.status;
 
 	if (status === null) {
-		redirect("/apply");
+		redirect(searchParams?.return_to ?? "/");
```

Comment:
Note for later: we should consider having the Choose your Character section anchored with an ID so we can navigate to something like `/#apply` in situations like these.